scripts/lib/CIME/namelist.py

```diff
@@ -899,9 +899,9 @@ def get_group_names(self):
         >>> Namelist().get_group_names()
         []
         >>> sorted(parse(text='&foo / &bar /').get_group_names())
-        [u'bar', u'foo']
+        ['bar', 'foo']
         """
-        return self._groups.keys()
+        return list(self._groups.keys())
```

Comment:
I'm wondering if the conversion to `list` here was done just to get the doctests to pass? If so, I'd say a better solution would be to convert the result to `list` in the doctests - keeping the production code returning an iterator in python3.